Sources/HTTPTypes/HTTPField.swift

```diff
@@ -181,8 +195,43 @@ extension HTTPField: CustomPlaygroundDisplayConvertible {
     }
 }
 
+extension HTTPField: Codable {
+    enum CodingKeys: String, CodingKey {
+        case name
+        case value
+        case indexingStrategy
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(self.name, forKey: .name)
+        try container.encode(self.rawValue._storage, forKey: .value)
+        if self.indexingStrategy != .automatic {
+            try container.encode(self.indexingStrategy.rawValue, forKey: .indexingStrategy)
+        }
+    }
+
+    private enum DecodingError: Error {
+        case invalidValue(String)
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let name = try container.decode(Name.self, forKey: .name)
+        let value = try container.decode(String.self, forKey: .value)
+        guard Self.isValidValue(value) else {
+            throw DecodingError.invalidValue(value)
+        }
+        self.init(name: name, uncheckedValue: ISOLatin1String(unchecked: value))
+        if let indexingStrategyValue = try container.decodeIfPresent(UInt8.self, forKey: .indexingStrategy),
+           let indexingStrategy = DynamicTableIndexingStrategy(rawValue: indexingStrategyValue) {
+            self.indexingStrategy = indexingStrategy
+        }
```

Comment:
If we encode the raw value now then we can never change the raw values or move to a different representation under the hood. Does it make sense to actually back the strategy by an internal enum with String based cases which we can then write out?

Reply:
This is not quite true: we just need to remember the old values we used. So long as we can guarantee that we'll be able to construct back from an int, we're good. I think we should be fine with this.

Reply:
I'm not super comfortable with exposing the underlying int either, but it's an advanced feature most people won't touch and it should be fairly easy to retain compatibility.